### ISSUE TITLE
[Feature] RouteLLM

### ISSUE THREAD

**Issue by joshuayao** [opea-project/GenAIComps#936]
### Priority

P1-Stopper

### OS type

Ubuntu

### Hardware type

Xeon-GNR

### Running nodes

Single Node

### Description

LLMs exist wide variation in their costs and capabilities. Very broadly, more capable models tend to be more expensive than less capable models. This leads to a dilemma when deploying LLMs in the real-world - routing all queries to the largest, most capable model leads to the highest-quality responses but can be expensive, while routing queries to smaller models can save costs but may result in lower-quality responses. 

LLM routing offers a solution to this, where each query is first processed by a system that decides which LLM to route it to. Ideally, all queries that can be handled by weaker models should be routed to these models, with all other queries routed to stronger models, minimizing cost while maintaining response quality.

We have the [RFC](https://github.com/opea-project/docs/pull/310) for addressing the problem.

**Comment by haim-barad** [comment]
The RFC that was opened is based on RouteLLM (plus more features). I hope we can expedite this.

**Comment by haim-barad** [comment]
This issue can be closed. The RFC was merged into the docs and there is a PR awaitingt merger for the router. 

Note - I don't have permissions to close this issue. Please do so - thanks.

**Comment by joshuayao** [comment]
> This issue can be closed. The RFC was merged into the docs and there is a PR awaitingt merger for the router.
> 
> Note - I don't have permissions to close this issue. Please do so - thanks.

I will close this issue after #1563 is merged.

**Comment by joshuayao** [comment]
I will close this if #1709 get merged.
